telegram/ext/_utils/webhookhandler.py

```diff
@@ -116,6 +123,19 @@ async def post(self) -> None:
         self._logger.debug("Webhook triggered")
         self._validate_post()
 
+        # verifying that the secret token is the one the user set when the user set one
+        if self.secret_token is not None:
+            self._logger.debug("Webhook has a secret token, trying to get and validate it")
+            token = self.request.headers.get("X-Telegram-Bot-Api-Secret-Token")
+            if not token:
+                self._logger.debug("Request did not include the token")
+                raise tornado.web.HTTPError(HTTPStatus.FORBIDDEN)
+            if token != self.secret_token:
+                self._logger.debug("Request had the wrong token")
```

Comment:
```suggestion
                self._logger.debug("Request had the wrong secrect token")
```

maybe also include the wrong token in the debug log?